syncstorage/src/error.rs

```diff
@@ -322,34 +260,17 @@ where
     seq.end()
 }
 
-macro_rules! impl_fmt_display {
-    ($error:ty, $kind:ty) => {
-        impl fmt::Display for $error {
-            fn fmt(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
-                fmt::Display::fmt(&self.kind, formatter)
-            }
-        }
-    };
-}
-
 impl_fmt_display!(ApiError, ApiErrorKind);
 
-macro_rules! from_error {
-    ($from:ty, $to:ty, $to_kind:expr) => {
-        impl From<$from> for $to {
-            fn from(inner: $from) -> $to {
-                $to_kind(inner).into()
-            }
+impl From<DbError> for ApiError {
+    fn from(db_error: DbError) -> Self {
+        Self {
+            status: db_error.status,
+            backtrace: db_error.backtrace.clone(),
```

Comment:
I won't hold up this PR any longer over this but we could avoid this clone -- maybe adding a `backtrace2` method that defers to the inner `DbError` + making the `ApiError::backtrace` field a private `Option<Backtrace>`.

`thiserror` seems to provide a nice way of automatically implementing this for the `std::Error::backtrace` method but that method's still unstable, unfortunately making many backtrace things like this a bit clunky.

Reply:
I think we could also use [mem::take](https://doc.rust-lang.org/stable/std/mem/fn.take.html) here, but I'm not sure if that's actually much better than clone (i.e. would memory be allocated for the newly-created default value?)

Reply:
The answer appears to be yes, as `Backtrace::default()` just calls `Backtrace::new()`, which just creates a backtrace :(